packages/core/src/treemap.ts

```diff
@@ -0,0 +1,104 @@
+import { ISpecProp, IOnboardingSpec, IOnboardingMessage, defaultOnboardingStages, EDefaultOnboardingStages, IOnboardingStage } from "./interfaces";
+import {getAnchor} from './utils';
+
+export interface IOnboardingTreemapSpec extends IOnboardingSpec {
+  chartTitle?: ISpecProp;
+  desc?: ISpecProp;
+  subDesc?: ISpecProp;
+  otherDesc?: ISpecProp;
+  interactingDesc?: ISpecProp;
+  maxValueDesc?: ISpecProp;
+  minValueDesc?: ISpecProp;
+  minValue?: ISpecProp;  
+  maxValue?: ISpecProp;
+}
+
+function generateMessages(spec: IOnboardingTreemapSpec, visElement: Element): IOnboardingMessage[] {
+  const analyzing = defaultOnboardingStages.get(EDefaultOnboardingStages.ANALYZING) as IOnboardingStage;
+  const reading = defaultOnboardingStages.get(EDefaultOnboardingStages.READING) as IOnboardingStage;
+  const interacting = defaultOnboardingStages.get(EDefaultOnboardingStages.USING) as IOnboardingStage;
+
+  const messages = [
+    {
+      anchor: getAnchor(spec.chartTitle, visElement),
+      requires: ['chartTitle'],
+      text: `The chart shows the ${spec.chartTitle?.value}.`,
+      title: 'Reading the chart',
+      onboardingStage: analyzing,
+      marker: {
+        id: "unique-marker-id-1"
+      }
+    },
+    {
+      anchor: getAnchor(spec.desc, visElement),
+      requires: ['desc'],
+      text: `The treemap visualization shows the breakdown of hierarchical data level by level.The size of each rectangle represents a quantitative value associated with each element in the hierarchy.`,
+      title: 'Reading the chart',
+      onboardingStage: reading,
+      marker: {
+        id: "unique-marker-id-2"
+      }
+    },
+    {
+      anchor: getAnchor(spec.subDesc, visElement),
+      requires: ['subDesc'],
+      text: `The size of each rectangle represents a quantitative value associated with each element in the hierarchy.`,
```

Comment:
This is a partly duplicate of the first message. Please remove the last sentence of the fist message.